src/pages/Player.vue

```diff
@@ -1,327 +1,68 @@
-      componentProperties: {}, // properties of the plio player
-      items: [], // holds the list of all items for this plio
-      itemResponses: [], // holds the responses to each item
-      videoSource: "youtube", // source for the video
-      watchTime: 0, // keeps a count of the watch time in seconds for the plio by the user
-      watchTimeIncrement: 0, // maintains the increase in watch time since the last time it was logged
-      currentItemIndex: null, // current item being displayed
-      markerClass: [
-        // class for the item marker displayed on top of the video slider
-        "bg-red-600",
-        "absolute",
-        "z-10",
-        "transform",
-        "translate",
-        "-translate-x-2/4",
-        "translate-y-4",
-        "py-1.5",
-        "px-1",
-        "bottom-full",
-        "pointer-events-none",
-        "rounded-md",
-      ],
-      playerHeight: 0, // height of the player - updated once the player is ready
-      lastCheckTimestamp: 0, // time in milliseconds when the last check for item pop-up took place
-      isFullscreen: false, // is the player in fullscreen
-      currentTimestamp: null, // tracks the current timestamp in the video
-      plioDBId: null, // id for this plio in the plio DB table
-      sessionDBId: null, // id for this session in the plio DB table
-      retention: [], // array to store video retention value
-      lastTimestampRetention: null, // last recorded timestamp in the retention array
-      toast: useToast(), // use the toast component
-      isModalMinimized: false, // whether the item modal is minimized or not
-      // styling class for the minimize button
-      maximizeButtonClass:
-        "bg-primary hover:bg-primary-hover p-1 pl-4 pr-4 sm:p-2 sm:pl-6 sm:pr-6 lg:p-4 lg:pl-6 lg:pr-6 rounded-md shadow-xl disabled:opacity-50 disabled:pointer-events-none",
+      isPlioLoaded: false, // whether plio has been loaded
+      isItemShown: false, // whether an item is being shown
+      previousPage: null, // the last page in the history
     };
   },
-  watch: {
-    isFullscreen(newIsFullscreen) {
-      // track the fullscreen status
-
-      if (newIsFullscreen) {
-        // trigger player fullscreen enter only if it is not already entered full screen
-        // by other modes like player controls or double click on the video.
-        if (!this.player.fullscreen.active) {
-          this.player.fullscreen.enter();
-        }
-      } else {
-        // trigger player full screen exit only if it is not already exited full screen
-        // by other modes like player controls or double click on the video.
-        if (this.player.fullscreen.active) {
-          this.player.fullscreen.exit();
-        }
-      }
-    },
+  /**
+   * set the previous page in history
+   */
+  beforeRouteEnter(_, from, next) {
+    next((vm) => {
+      vm.previousPage = from.path;
+    });
```

Comment:
Ideal to do it in router only na?
We had this discussion earlier to keep router logic separate